xbstrap/vcs_utils.py

```diff
@@ -17,6 +18,16 @@ class RepoStatus(Enum):
     OUTDATED = 2
 
 
+CHECKSUM_BLOCK_SIZE = 1048576
+HASHLIB_MAP = {
+    "md5": hashlib.md5(),
+    "sha1": hashlib.sha1(),
+    "sha256": hashlib.sha256(),
+    "sha512": hashlib.sha512(),
+    "blake2b": hashlib.blake2b(),
+}
```

Comment:
I'd rather store the class in this map. Then we can create a new csum object for each file, instead of re-using objects:
```python
Csum = HASHLIB_MAP.get(csum_type)
csum = Csum()
...
```

Reply:
ah, yes, indeed, this might even be a bug